core/amber/src/main/resources/python_udf/texera_udf_main.py

```diff
@@ -1,65 +1,85 @@
+import contextlib
 import importlib.util
 import logging
 import os
 import sys
 from datetime import datetime
 from pathlib import Path
 
+from loguru import logger
+
 from operators.texera_filter_operator import TexeraFilterOperator
 from operators.texera_map_operator import TexeraMapOperator
 from server.udf_server import UDFServer
 
 
-def init_root_logger(log_input_level: str,
-                     stream_log_level: str, stream_log_fmt: str, stream_datefmt: str,
-                     file_log_dir: str, file_log_level: str, file_log_fmt: str, file_datefmt: str) -> None:
+class InterceptHandler(logging.Handler):
+    def emit(self, record):
+        # Get corresponding Loguru level if it exists
+        try:
+            level = logger.level(record.levelname).name
+        except ValueError:
+            level = record.levelno
+
+        # Find caller from where originated the logged message
+        frame, depth = logging.currentframe(), 2
+        while frame.f_code.co_filename == logging.__file__:
+            frame = frame.f_back
+            depth += 1
+
+        logger.opt(depth=depth, exception=record.exc_info).log(level, record.getMessage())
+
+
+class StreamToLogger(object):
+
+    def __init__(self, level="INFO"):
+        self._level = level
+
+    def write(self, buffer):
+        for line in buffer.rstrip().splitlines():
+            logger.opt(depth=1).log(self._level, "[print] " + line.rstrip())
+
+    def flush(self):
+        pass
+
+
+def init_root_logger(stream_log_level: str, stream_log_fmt: str,
+                     file_log_dir: str, file_log_level: str, file_log_fmt: str) -> None:
     """
     initialize root logger with the given configurations
-    :param log_input_level: level to be captured by the root logger
     :param stream_log_level: level to be output to stdout/stderr
     :param stream_log_fmt: log format to stdout/stderr
-    :param stream_datefmt: log datefmt to stdout/stderr
     :param file_log_dir: log file directory
     :param file_log_level: level to be output to file
-    :param file_log_fmt: log format to file
-    :param file_datefmt: log datefmt to file
+    :param file_log_fmt: log format to fil
     :return:
     """
-
-    logger = logging.getLogger()
-    logger.setLevel(log_input_level)
-
+    logger.remove()
```

Comment:
why are we first removing and then adding a logger?

Reply:
to reset the configurations, see comments for details